src/Common/dataAccess/readCollection.ts

```diff
@@ -22,3 +26,51 @@ export async function readCollection(databaseId: string, collectionId: string):
   clearMessage();
   return collection;
 }
+
+export async function getMongoCollectionIndexTransformationProgress(
+  databaseId: string,
+  collectionId: string
+): Promise<number> {
+  let indexTransformationPercentage: number;
+  const clearMessage = logConsoleProgress(`Reading container ${collectionId}`);
+  try {
+    const response = await client()
+      .database(databaseId)
+      .container(collectionId)
+      .read({ populateQuotaInfo: true });
+
+    indexTransformationPercentage = parseInt(
+      response.headers[Constants.HttpHeaders.collectionIndexTransformationProgress] as string
+    );
+  } catch (error) {
+    logConsoleError(`Error while reading container ${collectionId}:\n ${JSON.stringify(error)}`);
+    logError(JSON.stringify(error), "ReadCollection", error.code);
+    sendNotificationForError(error);
+    throw error;
+  }
+  clearMessage();
+  return indexTransformationPercentage;
+}
+
+export async function readMongoDBCollectionThroughRP(
```

Comment:
Why are we exposing this? The idea of having `readCollection` was to abstract all of this away so the caller doesn't have to worry about the API specifics

Reply:
I can instead add a DefaultExperince based override within readCollection and not expose this method. will that work?

Reply:
Will move to a separate file since we will need both the readCollection through a client and through RP.